### PR TITLE
Corrections to flash driver documentation

### PR DESCRIPTION
**Pull request by networkfusion**
## Motivation and Context
The documentation and comments were out of date for STM32 flash driver. This corrects them.

Linked to: https://github.com/nanoframework/Home/issues/913

Note: the H7 driver is incorrect and support sill needs adding.

## How Has This Been Tested?
None required.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: targets/ChibiOS/_nf-overlay/stm32-internal-flash-driver.md [hunk 1]
```diff
@@ -2,12 +2,13 @@
 
 This driver enables **nanoFramework** to access the internal flash for STM32 parts taking care of the low level details and implementation specifics.
 
-In order to enable/use it the developer has to set HAL_USE_STM32_FLASH to TRUE on the halconf_nf.h provided in the reference implementations.
-_Note: to enable the driver the developer has to also enable the HAL_USE_COMMUNITY 'master switch' on that same file._
+In order to enable/use it the developer has to set HAL_NF_USE_STM32_FLASH to TRUE on the halconf_nf.h provided in the reference implementations.
 
 There are currently two versions of the driver:
```
Comment (josesimoes) [inline targets/ChibiOS/_nf-overlay/stm32-internal-flash-driver.md:7]:
```suggestion
The following versions of the driver are available:
```

File: CMake/Modules/FindChibiOS_H7_HAL.cmake [hunk 2]
```diff
@@ -52,7 +52,7 @@ endif()
 # WHEN ADDING A NEW CHIBIOS OVERLAY component add the include directory(ies) below 
 ####################################################################################
 # component STM32_FLASH
-list(APPEND CHIBIOS_HAL_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/targets/ChibiOS/_nf-overlay/os/hal/ports/STM32/LLD/FLASHv2)
+list(APPEND CHIBIOS_HAL_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/targets/ChibiOS/_nf-overlay/os/hal/ports/STM32/LLD/FLASHv3) ### TODO... Correct version???
```
Comment (josesimoes) [inline CMake/Modules/FindChibiOS_H7_HAL.cmake:55]: If we aren't doing anything with it right, just leave it as it is...

File: CMake/Modules/FindChibiOS_H7_HAL.cmake [hunk 3]
```diff
@@ -73,7 +73,7 @@ list(APPEND CHIBIOS_HAL_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/targets/ChibiOS/_nf-ove
 # WHEN ADDING A NEW CHIBIOS OVERLAY component add the source file(s) specific to this series below 
 ####################################################################################################
 # component STM32_FLASH
-list(APPEND CHIBIOS_HAL_SOURCES ${CMAKE_SOURCE_DIR}/targets/ChibiOS/_nf-overlay/os/hal/ports/STM32/LLD/FLASHv2/flash_lld.c)
+list(APPEND CHIBIOS_HAL_SOURCES ${CMAKE_SOURCE_DIR}/targets/ChibiOS/_nf-overlay/os/hal/ports/STM32/LLD/FLASHv3/flash_lld.c) ### TODO... Correct version???
```
Comment (josesimoes) [inline CMake/Modules/FindChibiOS_H7_HAL.cmake:76]: If we aren't doing anything with it right, just leave it as it is...

**Review by josesimoes — CHANGES_REQUESTED** [review]

File: CMake/Modules/FindChibiOS_H7_HAL.cmake (hunk 2 above)
Comment (networkfusion) [reply CMake/Modules/FindChibiOS_H7_HAL.cmake:55]: I have just raised an issue. I am happy to remove the change, but would raise an issue about it not working.

Comment (networkfusion) [reply CMake/Modules/FindChibiOS_H7_HAL.cmake:55]: But as it is definitely broken, wondering whether we should leave a comment regardless...

Comment (josesimoes) [reply CMake/Modules/FindChibiOS_H7_HAL.cmake:55]: There is no working target in STM32H7 series, so... This got here on a copy/past operation when starting work on adding one of those. With Azure RTOS close to become available, I wouldn't bother with this.

Comment (networkfusion) [reply CMake/Modules/FindChibiOS_H7_HAL.cmake:55]: Ok, in that case, we should remove the file?

Comment (networkfusion) [reply CMake/Modules/FindChibiOS_H7_HAL.cmake]: But anyway, reverted, so we can just call this PR a doc update.

**Review by josesimoes — APPROVED** [review]
Perfect! Thanks for fixing another bit! 👍🏻

**Pull request merged**